services/workflows-service/src/business-report/business-report.service.ts

```diff
@@ -160,27 +182,58 @@ export class BusinessReportService {
 
     await this.prisma.$transaction(
       async transaction => {
-        const businessCreatePromises = businessReportsRequests.map(async businessReportRequest => {
-          let business =
-            businessReportRequest.correlationId &&
-            (await this.businessService.getByCorrelationId(businessReportRequest.correlationId, [
-              projectId,
-            ]));
+        const businessesLookup = new Map<string, Business>();
+
+        const allCorrelationIds = new Set(
+          businessReportsRequests
+            .map(({ correlationId }) => correlationId)
+            .filter(correlationId => !!correlationId) as string[],
```

Comment:
```suggestion
            .filter(correlationId => Boolean(correlationId)) as string[],
```

Nit. May be a little more readable to cast explicitly, but up to you

Reply:
`.filter(Boolean)` is even better here.

Reply:
There is more Boolean(...)-like casts, so I agree with you. ESLint rule should be added to have just one type of casts.